selfie.c

```diff
@@ -9393,6 +9891,16 @@ void print_profile(uint64_t* context) {
     print_register_memory_profile();
   }
 
+  if (L1_CACHE_ENABLED) {
+    print_cache_statistic(get_cache_hits(L1_DCACHE), get_cache_misses(L1_DCACHE), "d");
+    println();
+
+    print_cache_statistic(get_cache_hits(L1_ICACHE), get_cache_misses(L1_ICACHE), "i");
```

Comment:
This reports 1 more fetch than the total number of executed instructions. Where's the bug? :)

Reply:
We found the reason for that (and why the statistics aren't "wrong" for all programs): If a load/store causes a page fault the page will be mapped and the same instruction will be executed again (this time successfully). This means that the instruction is fetched two times but `ic_load`/`ic_store` is only incremented one time.
In my opinion both, the number of icache accesses and the number of (successfully) executed instructions, are correct.

Reply:
Sounds good :)

Reply:
Check out my changes and please look for errors. It was a bit more work than I anticipated.

Reply:
The changes look good. I didn't spot anything that looks wrong.

Reply:
I don't see any problems with the changes either.